apps/api/src/app/environments-v1/usecases/outputRendererUseCases.ts

```diff
@@ -0,0 +1,144 @@
+// Base interface for all renderers
+import { render } from '@maily-to/render';
+import {
+  ChatRenderOutput,
+  EmailRenderOutput,
+  InAppRenderOutput,
+  PushRenderOutput,
+  RedirectTargetEnum,
+  SmsRenderOutput,
+  TipTapNode,
+} from '@novu/shared';
+import { z } from 'zod';
+import { ExpendEmailEditorSchemaUseCase } from '../../environments/render/email-schema-extender';
+
+class RenderCommand {
+  controlValues: Record<string, unknown>;
+}
+
+export class EmailOutputRendererUseCase {
+  async execute(renderCommand: RenderCommand): Promise<EmailRenderOutput> {
+    const parse = EmailStepControlSchema.parse(renderCommand.controlValues);
+    const schema = parse.emailEditor as TipTapNode;
+    const expandedSchema = new ExpendEmailEditorSchemaUseCase().execute({ schema });
+    const html = await render(expandedSchema);
+
+    return { subject: parse.subject, body: html };
+  }
+}
+
+// Concrete Renderer for Chat Preview
+export class ChatOutputRendererUseCase {
+  execute(renderCommand: RenderCommand): ChatRenderOutput {
+    const body = (renderCommand.controlValues.body as string) || 'Default chat message';
+
+    return { body };
+  }
+}
+
+// Concrete Renderer for SMS Preview
+export class SmsOutputRendererUseCase {
+  execute(renderCommand: RenderCommand): SmsRenderOutput {
+    const body = (renderCommand.controlValues.body as string) || 'Default SMS message';
+
+    return { body };
+  }
+}
+
+// Concrete Renderer for Push Notification Preview
+export class PushOutputRendererUseCase {
+  execute(renderCommand: RenderCommand): PushRenderOutput {
+    const subject = (renderCommand.controlValues.subject as string) || 'Default Push Notification Subject';
+    const body = (renderCommand.controlValues.body as string) || 'Default Push Notification Body';
+
+    return { subject, body };
+  }
+}
+
+// Concrete Renderer for In-App Message Preview
+export class InAppOutputRendererUseCase {
+  execute(renderCommand: RenderCommand): InAppRenderOutput {
+    console.log('InAppOutputRendererUseCase.execute', JSON.stringify(renderCommand, null, 2));
+    const inApp = InAppRenderOutputSchema.parse(renderCommand.controlValues);
+
+    return {
+      subject: inApp.subject as string,
```

Comment:
Are the type assertions required at this stage? Isn't the `inApp` typed?

Reply:
removed it, unnecessary